sdk/eventhub/Azure.Messaging.EventHubs/src/Core/PartitionResolver.cs

```diff
@@ -92,15 +95,26 @@ public virtual string AssignRoundRobin(string[] partitions)
         ///   the gateway hashing algorithm as closely as possible and should not be adjusted without careful
         ///   consideration.
         /// </remarks>
-        ///
+        [SkipLocalsInit]
         private static short GenerateHashCode(string partitionKey)
         {
             if (partitionKey == null)
             {
                 return 0;
             }
 
-            ComputeHash(Encoding.UTF8.GetBytes(partitionKey), seed1: 0, seed2: 0, out uint hash1, out uint hash2);
+            byte[] sharedBuffer = null;
+            var partitionKeySpan = partitionKey.AsSpan();
+            var partitionKeyByteLength = Encoding.UTF8.GetByteCount(partitionKeySpan);
+            Span<byte> hashBuffer = partitionKeyByteLength <= 256 ? stackalloc byte[partitionKeyByteLength] :
```

Comment:
I used 256 as a max size similar to what Json does here

https://github.com/dotnet/runtime/blob/ab02bf6745a80d991ccfbdac0f4e029b9d16f14b/src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.Unescaping.cs#L74-L76

since it is hard to come up with a good threshold

Reply:
FYI There was recently an interesting conversation that has also been captured in 

https://github.com/dotnet/docs/issues/28823

I did run some tests and if I switch the pattern the numbers look like

![image](https://user-images.githubusercontent.com/174258/160423271-bc702db2-763d-4502-9b3c-09930c046cb6.png)

but by adding `SkipLocalsInit` I'm now getting an impressive boost

![image](https://user-images.githubusercontent.com/174258/160423389-0226c8f6-e322-44ca-ba7f-981012eeded9.png)

Reply:
Or in other words I would like to bring back https://github.com/Azure/azure-sdk-for-net/pull/27598/commits/14912a3cfda8253debf92db9944c62a0d25ffcc2

Reply:
Brought the SkipLocalsInit back and aligned the stack alloc pattern to always go for the max size as suggested by the linked conversation